### PR TITLE
More general documentation

### PR DESCRIPTION
**Pull request by djmitche**
What is rendering? What's a context? How is all of this typically used?

### CODE REVIEW

File: docs/src/language.md [hunk 1]
```diff
@@ -5,3 +5,18 @@ The following sections describe the JSON-e language.
...
+
+A JSON-e template is rendered with a context to produce a result. The
+template, context, and result are all data structures containing strings,
+numbers, true/false, null, objects, and arrays -- the type of data structure
+you get from parsing JSON.
+
+The context is an object, giving values for variables at the top level of the
+template. JSON-e defines a few such variables, too, documented in the
+[Built-Ins](./built-ins.md) section. Some operators, such as `$let`, allow the
+template to define additional variables during rendering.
```
Comment (gregsdennis) [inline docs/src/language.md:19]: Is it with mentioning that locally-defined variable override those provided with the context?  ("Scope" might be a good word to use.)

Comment (djmitche) [reply docs/src/language.md]: Good point, thanks!

**Review by gregsdennis — APPROVED** [review]

File: docs/src/language.md [hunk 2]
```diff
@@ -5,3 +5,28 @@ The following sections describe the JSON-e language.
...
 irrelevant to its operation.
+
+## Rendering
+
+A JSON-e template is rendered with a context to produce a result. The template,
+context, and result are all data structures containing strings, numbers,
+true/false, null, objects, and arrays -- the type of data structure you get
+from parsing JSON.
+
+The context is an object, giving values for variables at the top level of the
```
Comment (gregsdennis) [inline docs/src/language.md:16]: > The context is an object

The playground doesn't error for a non-object context if the context is never referenced.

Comment (djmitche) [reply docs/src/language.md:16]: Ah,
```js
  if (typeof context !== 'object') {
```
is true for arrays (Array is a class, so an instance is an object).

Comment (djmitche) [reply docs/src/language.md:16]: Fixed in #481.

Comment (gregsdennis) [reply docs/src/language.md:16]: Ah... JavaScript!

**Pull request merged**